pkg/controller/core/workload_controller_test.go

```diff
@@ -527,6 +527,17 @@ func TestReconcile(t *testing.T) {
 				Message:   "Deactivated Workload \"ns/wl\" by reached re-queue backoffLimitCount",
 			}},
 		},
+		// This case would not work well due to fake client limitation,
+		// but we already verified this case in the integration test.
+		//"deactivated workload": {
+		//	workload: utiltesting.MakeWorkload("wl", "ns").
+		//		Active(false).
+		//		RequeueState(ptr.To[int32](1), ptr.To(metav1.NewTime(testStartTime.Add(1*time.Second).Truncate(time.Second)))).
+		//		Obj(),
+		//	wantWorkload: utiltesting.MakeWorkload("wl", "ns").
+		//		Active(false).
+		//		Obj(),
+		//},
```

Comment:
This test didn't work well for the same reason: https://github.com/kubernetes-sigs/kueue/pull/2063/files#r1589119722.